### ISSUE TITLE
Low accuracy using examples/llm/finetune/automodel.py

### ISSUE THREAD

**Issue by sravan500** [NVIDIA/NeMo#14170]
Hi Team,
I have been running peft using examples/llm/finetune/automodel.py script for finetuning 7b model.
I have a custom codebase using seqtoseqtrainer.
While accuracy is "high" with custom codebase compared to NeMo.
The training speed is very fast in Nemo Automodel.

Can you please confirm if I am running correct codebase?

Did any one used example scripts?

Waiting to hear from you.
Thanks in Advance.

**Comment by akoumpa** [comment]
Hi @sravan500, thanks for opening this issue.

I think it would hard for us to debug without access to data or the dataset class used. 

I was wondering if you would be able to share any part of the data, this could be just a few samples, and/or the dataset class. I want to make sure the dataset/formatting provides the correct data. Please feel free to email me at akoumparouli@nvidia.com if you don't want to share data publicly.

We are also moving this code-base to a new repo available [here](https://github.com/NVIDIA-NeMo/Automodel), I would suggest to check it out as well, as it's leaner and we'll be focusing our development there.

Looking forward to your response. Thank you.

**Comment by sravan500** [comment]
@akoumpa many thanks for responding.
I have only executed this code as is.
examples/llm/finetune/automodel.py

with command:
python automodel.py

No changes were made  during running.
It is using squad data module,whihc is the default.

My concern is to check ,if I have used the right code base(because it is there in examples)

**Comment by sravan500** [comment]
@akoumparouli
I see that the automodel.py code trains the model very fast.
Just curious to know if It is  because of the Nemo's custom FSDP2  implementation or the Pytorch lightning trainer.

Can I know what are the optimizations which are helping in faster training.

Thanks in Advance.
